pkg/util/push/push.go

```diff
@@ -17,14 +17,16 @@ func Handler(cfg distributor.Config, push func(context.Context, *client.WriteReq
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		compressionType := util.CompressionTypeFor(r.Header.Get("X-Prometheus-Remote-Write-Version"))
 		var req client.PreallocWriteRequest
-		req.Source = client.API
 		_, err := util.ParseProtoReader(r.Context(), r.Body, int(r.ContentLength), cfg.MaxRecvMsgSize, &req, compressionType)
 		logger := util.WithContext(r.Context(), util.Logger)
 		if err != nil {
 			level.Error(logger).Log("err", err.Error())
 			http.Error(w, err.Error(), http.StatusBadRequest)
 			return
 		}
+		if req.Source == 0 {
+			req.Source = client.API
+		}
```

Comment:
As per our [instructions](https://cortexmetrics.io/docs/contributing/#workflow), every commit must explain the changes made.  This is an extraordinary change, so needs a very good explanation.

Reply:
I'll send another PR to add comment here.